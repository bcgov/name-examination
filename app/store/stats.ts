import { getStats } from '~/util/namex-api'
import { getFormattedDateWithTime } from '~/util/date'
import { sortNameChoices } from '~/util'
import { type NameRequest } from '~/types'
import { defineStore } from 'pinia'

export const useStatsStore = defineStore('stats', () => {
  /**
   * Fetches paged statistics data based on the provided timespan and myStats flag.
   *
   * @param {string | null} timespan - The timespan for the statistics data. If null, the default timespan is used.
   * @param {boolean} myStats - A flag indicating whether to fetch statistics related to user's own stats.
   * @return {Promise<any>} A promise that resolves with the fetched statistics data.
   * @throws {Error} Throws an error if the fetching process fails.
   */
  async function getPagedStats (timespan: string | null, myStats: boolean): Promise<any> {
    const params = new URLSearchParams({
      timespan: timespan || '1',
      currentpage: '1',
      perpage: '1000'
    })
    if (myStats) params.append('myStats', 'True')

    try {
      const response = await getStats(params)

      if (response.ok) {
        const jsonData = await response.json() // Parse response JSON

        if (jsonData.numRecords >= 0) {
          const requestsData = jsonData.nameRequests as Array<NameRequest>

          const processedData = requestsData.map((statsData:any) => {
            statsData.names = sortNameChoices(statsData.names)
            if (statsData.comments.length === 1 && statsData.comments[0].timestamp) {
              statsData.comments[0].timestamp = getFormattedDateWithTime(statsData.comments[0].timestamp)
            }
            statsData.lastUpdate = getFormattedDateWithTime(statsData.lastUpdate)
            statsData.names.forEach((name:any) => {
              if (name.decision_text) {
                name.decision_text = name.decision_text.split(/\n\n/)
              }
            })
            return statsData
          })
          return {
            nameRequests: processedData,
            numRecords: jsonData.numRecords
          }
        }
      } else {
        throw new Error('Network response was not ok')
      }
    } catch (error) {
      console.error('There was a problem with your fetch operation:', error)
      throw error // Rethrow the error to handle it elsewhere if needed
    }
  }

  return {
    getPagedStats
  }
})
