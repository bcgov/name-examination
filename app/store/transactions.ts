import {
  ConsentFlag,
  type EntityTypeCode,
  type RequestActionCode,
  type RequestTypeCode,
} from '~/enums/codes'
import { Status } from '~/enums/nr-status'
import {
  type NameChoice,
  type NameRequest,
  type TransactionEntry,
  type Transactions,
} from '~/types'
import { emitter } from '~/util/emitter'
import { getNameRequest, getTransactions } from '~/util/namex-api'
import { toMappedRequestType } from '~/util/request-type'
import requestTypes from '~/data/request_types.json'
import { sortNameChoices } from '~/util'

export const useTransactions = defineStore('transactions', () => {
  const nr = ref<NameRequest>()
  const transactions = ref<Array<TransactionEntry>>([])
  const loadingNr = ref(false)
  const loadingTransactions = ref(false)

  async function loadNr(nrNumber: string) {
    loadingNr.value = true
    const nrResponse = await getNameRequest(nrNumber)
    nr.value = await nrResponse.json()
    loadingNr.value = false
  }

  async function loadTransactions(nrNumber: string) {
    loadingTransactions.value = true
    const transactionsResponse = await getTransactions(nrNumber)
    const transactionsJson = (await transactionsResponse.json()) as Transactions
    transactionsJson.transactions.forEach(t => sortNameChoices(t.names))
    transactions.value = transactionsJson.transactions
    loadingTransactions.value = false
  }

  async function initialize(nrNumber: string) {
    try {
      await loadNr(nrNumber)
      await loadTransactions(nrNumber)
    } catch (e) {
      emitter.emit('error', {
        title: 'Transactions Error',
        message: 'Failed to retrieve transactions data.',
      })
    } finally {
      loadingNr.value = false
      loadingTransactions.value = false
    }
  }

  function getStatusDisplay(status: Status | null, names: Array<NameChoice>) {
    if (!status) return 'N/A'

    let displayState = status as string
    if (status === Status.Conditional) {
      displayState = 'CONDITIONAL APPROVED'
    } else if (status === Status.Consumed && names && names.length > 0) {
      const approvedName = names.find((name) =>
        [Status.Approved, Status.Condition].includes(name.state)
      )
      displayState =
        approvedName?.state === Status.Condition
          ? 'CONDITIONAL APPROVED'
          : 'APPROVED'
      displayState += ` / Used for ${approvedName?.corpNum}`
    }
    return displayState
  }

  function getRequestTypeDisplay(
    requestType: RequestTypeCode,
    action: RequestActionCode
  ) {
    if (!nr.value) return 'N/A'
    const mapped = toMappedRequestType(
      requestType,
      action,
      nr.value.entity_type_cd
    )
    return requestTypes.find((r) => r.value == mapped)?.text || 'N/A'
  }

  function getConsentDisplay(date: string | null, consentFlag: ConsentFlag) {
    return date
      ? 'Required. Received.'
      : consentFlag === ConsentFlag.Required
      ? 'Required. Not yet received.'
      : 'Not required'
  }

  return {
    nr,
    transactions,
    loadingNr,
    loadingTransactions,
    initialize,
    getStatusDisplay,
    getRequestTypeDisplay,
    getConsentDisplay,
  }
})
