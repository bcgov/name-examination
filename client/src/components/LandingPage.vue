<!--eslint-disable -->
<template>
  <div class="container-fluid homepage-container">
    <div class="row">
      <div class="col">


  <div v-if="!auth">
    <h2 >Your authorization is missing or has expired. Please login.</h2>
  </div>
  <div v-else>
    <h3>Welcome to Name X!</h3>

    <p>Canada’s most modern semi - automated way to examine business name requests. Automated indicators showing you:</p>
    <p>
      <ul>
        <li><b>Conflicts</b> –Searches the name against related industry categories, synonyms, word substitutions ie: 9, nine) and more.
            The conflict search is only as good as the information in it, so if you don’t see something come back
            in the search you should check the SOLR admin tool to see if it is in the correct industry category or other search options
        <li><b>Condition</b> – This gives you the messages to clients and internal information based on condition of words or phrases ie: Doctor, BC etc…</li>
        <li><b>Trademarks</b> – Searches the Canadian Trademarks database and shows you active trademarks related to the name request</li>
        <li><b>History</b> – If a similar name has been approved or rejected previously (use this to tell you if they are holding a name to long)</li>
      </ul>
    </p>
    <p><i>Enjoy the ride!</i></p>
    <br/>

    <div class="card">
      <div class="card-body">
        <h2>Current status on {{todayStr}}</h2>
        <br/>
        <p>Not Examined: <b>{{statsData.draft.response.numFound}}</b><br/></p>
        <p>Hold: <b>{{statsData.hold.response.numFound}}</b><br/></p>
      </div>
    </div>
    <br/><br/>

    <span>
    <h2><i>Manual Search Tips:</i></h2>
    <p>Looks for all matching names including synonyms by default.</p><br/>
      <p>+&lt;<i>word</i>&gt; means must contain this word or its synonyms</p>
      <p>+"&lt<i>word1</i>&gt;&lt;<i>word2</i>&gt;" means must include both words or its synonyms</p>
      <p>-&lt<i>word</i>&gt; means remove this word for matching and all its synonyms</p>
      <p>-"&lt<i>word 1</i>&gt;&lt;<i>word 2</i>&gt;" means remove this word phrase and all its synonyms for matching.</p>
      <p>@&lt<i>word</i>&gt; means only use this word do not apply synonyms in matching.</p>
      <p>@"&lt<i>word 1</i>&gt;&lt;<i>word 2</i>&gt;" means only use this word phrase for these words in the name and do not substitute synonyms.</p>
      <p>? substitute one or more letters in a word. Eg. TRE??? – substitute three letters with any letters.</p>
      <p>* Substitute a portion of a word.</p>
    </span>

  </div>

      </div>
    </div>
  </div>

</template>

<script>
/* eslint-disable */

import StdHeader from "./application/sections/StdHeader";

export default {
  components: {StdHeader},
  name: 'LandingPage',
    watch: {
      auth: function (val) {
        console.log("Auth=" + val)
          if(val){ this.getCurrentStats() }
      }
    },
    computed: {
      auth() {
        return this.$store.getters.isAuthenticated
      },
      todayStr() {
        var months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
        var days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
        var d = new Date();
        return days[d.getDay()] + ", " + months[d.getMonth()] + " " + d.getDate() + " " + d.getFullYear()
      },
      statsData() {
        return this.$store.getters.statsDataJSON;
      }
    },
    methods: {
      //states: ['ALL', 'HOLD', 'INPROGRESS', 'DRAFT', 'EXPIRED', 'CANCELLED', 'APPROVED', 'CONDITIONAL', 'REJECTED'],
      getCurrentStats() {
        this.$store.statsDataJSON=null
        console.log("Get Statistics")
        //var nStates = ['hold', 'draft', 'expired', 'cancelled', 'approved', 'conditional', 'rejected']
        var nStates = ['hold', 'draft']
        var vm = this
        nStates.forEach( function(stateCd) {
          vm.$store.dispatch('getStatsDataJSON',stateCd);
        });
      }
    }
  }
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
  .homepage-container {
    padding-left: 50px;
    padding-right: 50px;
  }

  h1 {
    font-weight: bold;
    font-size: medium;
    padding-top: 1ch;
  }
  h2 {
    font-weight: bold;
    font-size: large;
  }
  h3 {
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
    font-weight: bold;
    font-size: 2.25em;
    color: black;
    padding-bottom: 1ch;
  }

  p {
    font-size: medium;
  }

  ul {
    padding-top: 1ch;
  }

  li{
    padding-bottom: 1ch;
  }

  a {
    color: #42b983;
  }

</style>
