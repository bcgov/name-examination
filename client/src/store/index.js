/* eslint-disable */
import Vue from 'vue'
import Vuex from 'vuex'
import axios from '../axios-auth'
import globalAxios from 'axios'
import router from '@/router'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    nrNum: null,
    authorized: null,
    userId: null,
    email: null,
    kctoken: null,
    nr_conflict: null,
    user_role: null,
    currentChoice: null,
    currentName: null,
    currentState: null,
    is_editing: false,
    nrData: null,
    details: null,
   // formData: {
      compInfo: {
        nrNumber: null,
        compNames: {
          compName1: null,
          compName2: null,
          compName3: null
        },
        requestType: null,
      },
      applicantInfo: {
        applicantname: {
          firstName: null,
          lastname: null
        },
        contactInfo: {
          address: null,
          contactName: null,
          phone: null,
          email: null,
          fax: null
        },
      },
      additionalCompInfo: {
        jurisdiction: null,
        natureOfBussiness: null,
        nuans: null,
        sk_name: null,
        nr_status: null
      },
      examiner: null,
      priority: null,
      reSubmission: {
        reSubmissionYN: null,
        linkedNR: null
      },
      reservationCount: null,
      expiryDate: null,
  //  },
    issueText: null,
    issue: {
        issue_Match: null,
        issue_Consent: null,
        issue_TradeMark: null,
        issue_History: null,
        issue_Format: null,
        issue_Match_Text: null,
        issue_Consent_Text: null,
        issue_TradeMark_Text: null,
        issue_History_Text: null,
        issue_Format_Text: null
      }
  },

  mutations: {
    authUser (state, userData) {
      state.kctoken = userData
      //state.kctoken = userData.client_session
      //state.userRole = userData.user_role
      //state.idToken = userData.client_session
    },

    clearAuthData (state) {
      state.kcToken = null
      state.userId = null
    },

    setIssueText (state,issue) {
      switch (issue) {
        case "M1":
          state.issueText = state.issue.issue_Match_Text;
          break;
        case "M2":
          state.issueText = state.issue.issue_Consent_Text;
          break;
        case "M3":
          state.issueText = state.issue.issue_TradeMark_Text;
          break;
        case "M4":
          state.issueText = state.issue.issue_History_Text;
          break;
        case "M5":
          state.issueText = state.issue.issue_Format_Text;
          break;
        default:
          state.issueText = state.issue.issue_Match_Text;
      }
    },

    loadpostgresNo(state, postgresData) {
        state.compInfo.nrNumber = postgresData.nameRequest
    },

    loadCompanyInfo(state, dbcompanyInfo) {
      console.log('Loading Company Info into state')

      state.nrData = dbcompanyInfo

      if (dbcompanyInfo.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of dbcompanyInfo.names) {
          switch (record.choice) {
            case 1:
              state.compInfo.compNames.compName1 = record.name
              break;
            case 2:
              state.compInfo.compNames.compName2 = record.name
              break;
            case 3:
              state.compInfo.compNames.compName3 = record.name
              break;
            default:
              console.log('Error with company name structure')
              return
          }
        }
      }

      state.compInfo.requestType = dbcompanyInfo.requestTypeCd
      state.applicantInfo.applicantname.firstName = dbcompanyInfo.applicant
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.applicantInfo.contactInfo.contactName = dbcompanyInfo.contact
      state.applicantInfo.contactInfo.phone = dbcompanyInfo.phoneNumber
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.additionalCompInfo.jurisdiction = dbcompanyInfo.xproJurisdiction
      state.additionalCompInfo.natureOfBussiness = dbcompanyInfo.natureBusinessInfo
      state.additionalCompInfo.nuans = dbcompanyInfo.nuansNum
      state.additionalCompInfo.sk_name = dbcompanyInfo.skPartner
      state.additionalCompInfo.nr_status = dbcompanyInfo.state
      state.examiner = dbcompanyInfo.userId
      state.priority = dbcompanyInfo.priorityCD
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.expiryDate = dbcompanyInfo.expiryDate
    },

    loadCompanyIssues(state, dbcompanyIssues) {
      //state.issue.issue_Match = dbcompanyIssues.isue_Match
      //state.issue.issue_Match_Text = dbcompanyIssues.issue_Match_Text
      //state.issue.issue_Consent = dbcompanyIssues.issue_Consent
      //state.issue.issue_Consent_Text = dbcompanyIssues.issue_Consent_Text
      //state.issue.issue_TradeMark = dbcompanyIssues.issue_TradeMark
      //state.issue.issue_TradeMark_Text = dbcompanyIssues.issue_TradeMark_Text
      //state.issue.issue_History = dbcompanyIssues.issue_History
      //state.issue.issue_History_Text = dbcompanyIssues.issue_History_Text
      //state.issue.issue_Format = dbcompanyIssues.issue_Format
      //state.issue.issue_Format_Text = dbcompanyIssues.issue_Format_Text
    },

    update_nrData(state){

      if(state.nrData.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of state.nrData.names) {
          switch (record.choice) {
            case 1:
              state.nrData.names[0].name = state.compInfo.compNames.compName1
              break;
            case 2:
              state.nrData.names[0].name = state.compInfo.compNames.compName2
              break;
            case 3:
              state.nrData.names[0].name = state.compInfo.compNames.compName3
              break;
          }
        }
      }

      state.nrData.requestTypeCd = state.compInfo.requestType
      state.nrData.applicant = state.applicantInfo.applicantname.firstName
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.nrData.contact = state.applicantInfo.contactInfo.contactName
      state.nrData.phoneNumber =  state.applicantInfo.contactInfo.phone
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.nrData.xproJurisdiction = state.additionalCompInfo.jurisdiction
      state.nrData.natureBusinessInfo = state.additionalCompInfo.natureOfBussiness
      state.nrData.nuansNum = state.additionalCompInfo.nuans
      state.nrData.skPartner = state.additionalCompInfo.sk_name
      //state.nrData.state =  state.additionalCompInfo.nr_status
      state.nrData.userId = state.examiner
      state.nrData.priorityCD = state.priority
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.nrData.expiryDate = state.expiryDate
    },

    saveDetail(state,detail){
      state.details = detail
    }
  },

  actions: {
    kcauth ({commit, dispatch, state}) {
      const kc = localStorage.getItem('KEYCLOAK_TOKEN')
      state.kctoken = kc
      //console.log(authData)
      //state.user_role =  authData.realm_access.roles,
      state.user_role = ''
      state.idToken = kc
      localStorage.setItem('kctoken',state.kctoken)
      localStorage.setItem('user_role',state.user_role)

      if (state.kctoken){
        console.log('KC Authorized user roles')
        commit('authUser', {
          //user_role: state.user_role,
          kctoken: state.kctoken
        })
      }
    },

    logout ({commit,state}) {

      commit('clearAuthData')

      localStorage.removeItem('expirationDate')
      localStorage.removeItem('token')
      localStorage.removeItem('user_role')
      localStorage.removeItem('userId')
      localStorage.removeItem('email')
      localStorage.removeItem('COMPINFO')

      localStorage.removeItem('kctoken')
      localStorage.removeItem('KEYCLOAK_TOKEN')
      localStorage.removeItem('AUTHORIZED')
      localStorage.removeItem('USERNAME')

    },

    tryAutoLogin ({commit}) {
    },

    storeUser ({commit, state}, userData) {
      if (!state.idToken) {
        return
      }
      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
        .then(res => console.log(res))
        .catch(error => console.log(error))
    },

    fetchUser ({commit, state}) {
      if (!state.idToken) {
        return
      }
      globalAxios.get('/users.json' + '?auth=' + state.idToken)
        .then(res => {
          console.log(res)
          const data = res.data
          const users = []
          for (let key in data) {
            const user = data[key]
            user.id = key
            users.push(user)
          }
          console.log(users)
          commit('storeUser', users[0])
        })
        .catch(error => console.log(error))
    },

    setDetails({commit, state}) {
     var detail = state.details?null:"1";
     commit('saveDetail',detail)
    },

    selectNameIssue ({commit, dispatch, state}, divID) {
      console.log('action: selected issue control:' + divID)
      localStorage.setItem('issueText',divID )
      commit('setIssueText', divID)
    },

    getpostgrescompNo ({commit, dispatch, state}) {
      console.log('action: select next company number from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/queues/@me/oldest'
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp No Response:' + response)
        commit('loadpostgresNo',response.data)
      })
    },

    getpostgrescompInfo ({commit},nrNumber) {
      console.log('action: getting data for company number: ' + nrNumber + ' from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + nrNumber
      console.log('URL:' + url)
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp Info Response:' + response.data)
        localStorage.setItem('COMPINFO',response.data)
        commit('loadCompanyInfo',response.data)
      })
      .catch(error => console.log('ERROR: ' + error))
    },

    updateNRState ({state},nrState) {
      console.log('Updating Examination state for number: ' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber

      axios.patch(url,{headers: {Authorization: `Bearer ${myToken}`}},{"state": nrState} )
           .then(function(response){
                console.log('state updated to ' + nrState + ' for ' + state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
      },

    nameApproved( {commit, state}) {
      console.log('Name Approved for ' + state.compInfo.nrNumber + ", " + state.currentName)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'APPROVED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url, state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name approved for ' + state.compInfo.nrNumber)
           })
           .catch(error => console.log('ERROR: ' + error))
    },

    nameRejected({commit, state}) {
      console.log('Name Rejected for NR #' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'REJECTED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url,state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name rejected for ' +  state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
    }
  },

  getters: {
    is_editing(state) {
      return state.is_editing
    },
    user(state) {
      return state.user
    },
    email(state) {
      return state.email
    },
    issueText(state) {
      return state.issueText
    },
    isAuthenticated(state) {
      return localStorage.getItem("AUTHORIZED")
    },
    nrNumber(state) {
      return state.compInfo.nrNumber
    },
    compName1(state) {
     return state.compInfo.compNames.compName1
    },
    compName2(state) {
      return state.compInfo.compNames.compName2
    },
    compName3(state) {
      return state.compInfo.compNames.compName3
    },
    requestType(state) {
      return state.compInfo.requestType
    },
    firstName(state) {
      return state.applicantInfo.applicantname.firstName
    },
    lastName(state) {
      return state.applicantInfo.applicantname.lastname
    },
    address(state) {
      return state.applicantInfo.contactInfo.address
    },
    conEmail(state) {
      return  state.applicantInfo.contactInfo.email
    },
    contactName(state) {
      return state.applicantInfo.contactInfo.contactName
    },
    phone(state) {
      return state.applicantInfo.contactInfo.phone
    },
    fax(state) {
      return state.applicantInfo.contactInfo.fax
    },
    jurisdiction(state) {
      return  state.additionalCompInfo.jurisdiction
    },
    natureOfBusiness(state) {
      return state.additionalCompInfo.natureOfBussiness
    },
    nuans(state) {
      return state.additionalCompInfo.nuans
    },
    sk_name(state) {
      return state.additionalCompInfo.sk_name
    },
    nr_status(state) {
      return state.additionalCompInfo.nr_status
    },
    examiner(state) {
      return state.examiner
    },
    priority(state) {
      return state.priority
    },
    reSubmissionYN(state) {
      return state.reSubmission.reSubmissionYN
    },
    linkedNR(state) {
      return state.reSubmission.linkedNR
    },
    reservationCount(state) {
      return state.reservationCount
    },
    expiryDate(state) {
      return state.expiryDate
    },
    issue_Match(state) {
      return state.issue.issue_Match
    },
    issue_Match_Text(state) {
      return state.issue.issue_Match_Text
    },
    issue_Consent(state) {
      return state.issue.issue_Consent
    },
    issue_Consent_Text(state) {
      return state.issue.issue_Consent_Text
    },
    issue_TradeMark(state) {
      return state.issue.issue_TradeMark
    },
    issue_TradeMark_Text(state) {
      return state.issue.issue_TradeMark_Text
    },
    issue_History(state) {
      return state.issue.issue_History
    },
    issue_History_Text(state) {
      return state.issue.issue_History_Text
    },
    issue_Format(state) {
      return state.issue.issue_Format
    },
    issue_Format_Text(state) {
      return state.issue.issue_Format_Text
    },
    details(state) {
      return state.details
    },
    addInfo(state) {
      return state.details
    }
  }
})
