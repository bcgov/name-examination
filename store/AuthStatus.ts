import { defineStore } from 'pinia';

export const useAuthStore = defineStore('auth', {
  state: () => ({
    auth: false,
  }),
  actions: {
    toggleAuth() {
      this.auth = !this.auth;
    },
  },
});
