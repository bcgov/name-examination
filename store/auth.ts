import ConfigHelper from '../util/config-helper'
import KeycloakServices from '../public/keycloak/keycloak'
import { SessionStorageKeys } from '../util/constants'
import { defineStore } from 'pinia'

interface AuthState {
  token: string;
  idToken: string;
  refreshToken: string;
  kcGuid: string;
  loginSource: string;
}

export const useAuthStore = defineStore({
  id: 'auth',
  state: (): AuthState => ({
    token: '',
    idToken: '',
    refreshToken: '',
    kcGuid: '',
    loginSource: ''
  }),
  getters: {
    isAuthenticated (state): boolean {
      return !!state.token
    },
    keycloakGuid (state): string {
      return state.kcGuid || KeycloakServices.getUserInfo().keycloakGuid
    },
    currentLoginSource (state): string {
      return state.loginSource || KeycloakServices.getUserInfo().loginSource
    }
  },
  actions: {
    setKCToken (token: string): void {
      this.token = token
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakToken, token)
    },
    setIDToken (idToken: string): void {
      this.idToken = idToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakIdToken, idToken)
    },
    setRefreshToken (refreshToken: string): void {
      this.refreshToken = refreshToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakRefreshToken, refreshToken)
    },
    setKCGuid (kcGuid: string): void {
      this.kcGuid = kcGuid
    },
    setLoginSource (loginSource: string): void {
      this.loginSource = loginSource
    },
    clearSession (): void {
      // Reset all state properties related to the user's session
      this.token = ''
      this.idToken = ''
      this.refreshToken = ''
      this.kcGuid = ''
      this.loginSource = ''

      // Clear the session storage values
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakIdToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakRefreshToken)
    },
    syncWithSessionStorage (): void {
      const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken) || ''
      const idToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakIdToken) || ''
      const refreshToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakRefreshToken) || ''

      this.token = token
      this.idToken = idToken
      this.refreshToken = refreshToken
    }
  }
})

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useAuthStore, import.meta.hot))
}
