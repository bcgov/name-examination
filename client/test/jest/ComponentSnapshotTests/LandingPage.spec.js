/* eslint-disable */
import { shallowMount } from '@vue/test-utils';
import LandingPage from '@/components/LandingPage';
import store from '@/store'

describe('LandingPage.vue Snapshot', () => {

    let component;
    beforeEach(() => {
        component = shallowMount(LandingPage, {store: store});
    });

   it("renders a LandingPage component", () => {
     expect(component.element).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
