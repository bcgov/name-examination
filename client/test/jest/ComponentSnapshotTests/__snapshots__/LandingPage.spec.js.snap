// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage.vue Snapshot renders a LandingPage component 1`] = `
<div
  class="container-fluid homepage-container"
>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div>
        <h2>
          Your authorization is missing or has expired. Please login.
        </h2>
      </div>
    </div>
  </div>
</div>
`;
